### DISCUSSION TITLE
Browser back and forward buttons do not work on GitHub code browser

### DISCUSSION THREAD

**Discussion by mwisnicki** [community/community#75889] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Browser navigation has been completely broken for a while.

### Steps
1. Open https://github.com/community/.github
2. Click `workflow-templates`
3. Using file tree on the left click `profile/README.md`
4. Press back button

### Expected
I see previous page

### Actual
Browser location bar reverts to https://github.com/community/.github/tree/main/workflow-templates but I'm still seeing readme

This has been going on for a while.
Back/forward buttons work less than 50% of the time.

**Comment by esiqveland** [comment]
The inconsistent issues with the back button (browser history) on github is driving me insane. Especially when you are trying to go through notifications.

**Reply by Hacksore** [reply]
Same here! This has to be related to a lot of the react rollouts they have done.

Hitting back and landing two pages back is a miserable experiencing 🫤.

**Comment by HalfWhitt** [comment]
I'm still experiencing this... seems particularly prominent on mobile, for whatever reason. Very irritating.
